### ISSUE TITLE
Problem with buildup: string no output variable specified

### ISSUE THREAD

**Issue by miroi** [robertodr/for_id#1]
Hi,

can not buildup for_id on IBM AIX:

```
ilias@147.213.80.175:~/work/programming/miro_ilias_programming/test_c++_template_ibm-aix/test-for_id-funct/for_id/build/.cmake ..-- Could NOT find Boost
-- Boost 1.54.0 not found. The pre-packaged version will be built.
--   Libraries to be built: filesystem, system
--   Toolset to be used: gcc
CMake Error at external/CMakeLists.txt:43 (string):
  string no output variable specified


-- Configuring incomplete, errors occurred!
ilias@147.213.80.175:~/work/programming/miro_ilias_programming/test_c++_template_ibm-aix/test-for_id-funct/for_id/build/.

```

**Comment by robertodr** [comment]
@miroi Simplified Boost recognition (only the headers are needed) Should build up now.

**Comment by miroi** [comment]
Hi, 
again problen, with Boost:

```
ilias@147.213.80.175:~/work/programming/miro_ilias_programming/test_c++_template_ibm-aix/test-for_id-funct/for_id/build/.cmake ..
-- The CXX compiler identification is GNU 4.8.1
-- Check for working CXX compiler: /gpfs/home/federic/opt/freeware/bin/c++
-- Check for working CXX compiler: /gpfs/home/federic/opt/freeware/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Could NOT find Boost
-- Configuring done
-- Generating done
-- Build files have been written to: /gpfs/home/ilias/work/programming/miro_ilias_programming/test_c++_template_ibm-aix/test-for_id-funct/for_id/build
ilias@147.213.80.175:~/work/programming/miro_ilias_programming/test_c++_template_ibm-aix/test-for_id-funct/for_id/build/.m
Scanning dependencies of target example.x
[ 50%] Building CXX object CMakeFiles/example.x.dir/example.cpp.o
In file included from /gpfs/home/ilias/work/programming/miro_ilias_programming/test_c++_template_ibm-aix/test-for_id-funct/for_id/example.cpp:7:0:
/gpfs/home/ilias/work/programming/miro_ilias_programming/test_c++_template_ibm-aix/test-for_id-funct/for_id/src/ForId.hpp:7:34: fatal error: boost/lexical_cast.hpp: No such file or directory
 #include <boost/lexical_cast.hpp>
                                  ^
compilation terminated.
CMakeFiles/example.x.dir/build.make:57: recipe for target 'CMakeFiles/example.x.dir/example.cpp.o' failed
make[2]: *** [CMakeFiles/example.x.dir/example.cpp.o] Error 1
CMakeFiles/Makefile2:63: recipe for target 'CMakeFiles/example.x.dir/all' failed
make[1]: *** [CMakeFiles/example.x.dir/all] Error 2
Makefile:75: recipe for target 'all' failed
make: *** [all] Error 2
ilias@147.213.80.175:~/work/programming/miro_ilias_programming/test_c++_template_ibm-aix/test-for_id-funct/for_id/build/.
```

**Comment by robertodr** [comment]
Fixed by commit 667b2671a751cdf9d61bc1366efbf744fed8c260
